pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+        origin = vertices[0]
+        a_end = []
+        b_end = []
+        tol = 1e-10
+        for v in vertices[1:]:
+            edge1 = GeometryOperators.v_points(origin, v)
+            for v2 in vertices[1:]:
+                if v2 != v:
+                    edge2 = GeometryOperators.v_points(origin, v2)
+                    if abs(GeometryOperators.v_dot(edge1, edge2)) < tol:
+                        a_end = v
+                        b_end = v2
+                        break
+            if a_end:
+                break
+        if not a_end:
+            a_end = vertices[1]
+            b_end = vertices[2]
+            return False, (origin, a_end, b_end)
+        return True, (origin, a_end, b_end)
+
+    @pyaedt_function_handler()
+    def cover_lines(self, selection):
+        """Cover a closed line and transform it to sheet.
+
+        Parameters
+        ----------
+        selection : str, int
+            Polyline object to cover.
+        Returns
+        -------
+        bool
+
+        References
+        ----------
+
+        >>> oEditor.CoverLines
+        """
+        obj_to_cover = self.convert_to_selections(selection, False)
+        self.oeditor.CoverLines(["NAME:Selections", "Selections:=", obj_to_cover, "NewPartsModelFlag:=", "Model"])
+        return True
+
+    @pyaedt_function_handler()
+    def cover_faces(self, selection):
+        """Cover a sheet.
+
+        Parameters
+        ----------
+        selection : str, int
+            Sheet object to cover.
+        Returns
+        -------
+        bool
+
+        References
+        ----------
+
+        >>> oEditor.CoverLines
+        """
+        obj_to_cover = self.convert_to_selections(selection, False)
+        self.oeditor.CoverSurfaces(["NAME:Selections", "Selections:=", obj_to_cover, "NewPartsModelFlag:=", "Model"])
+        return True
+
+    @pyaedt_function_handler()
+    def create_coordinate_system(
+        self,
+        origin=None,
+        reference_cs="Global",
+        name=None,
+        mode="axis",
+        view="iso",
+        x_pointing=None,
+        y_pointing=None,
+        psi=0,
+        theta=0,
+        phi=0,
+        u=None,
+    ):
+        """Create a coordinate system.
+
+        Parameters
+        ----------
+        origin : list
+            List of ``[x, y, z]`` coordinates for the origin of the
+            coordinate system.  The default is ``None``, in which case
+            ``[0, 0, 0]`` is used.
+        reference_cs : str, optional
+            Name of the reference coordinate system. The default is
+            ``"Global"``.
+        name : str
+            Name of the coordinate system. The default is ``None``.
+        mode : str, optional
+            Definition mode. Options are ``"view"``, ``"axis"``,
+            ``"zxz"``, ``"zyz"``, and ``"axisrotation"``. The default
+            is ``"axis"``.  Enumerator ``pyaedt.generic.constants.CSMODE`` can be used.
```

Comment:
```suggestion
            Definition mode. Options are ``"axis"``, ``"axisrotation"``,
            ``"view"``, ``"zxz"``, and ``"zyz"`` The default
            is ``"axis"``.  You can also use the ``pyaedt.generic.constants.CSMODE`` enumerator.
```
Options should be listed alphabetically.